src/mcpbr/reproducibility.py

```
@@ -112,11 +112,11 @@ def _collect_packages() -> dict[str, str]:
         from importlib.metadata import distributions
 
         for dist in distributions():
-            name = dist.metadata.get("Name", "")
-            version = dist.metadata.get("Version", "")
+            name = dist.metadata["Name"] or ""
+            version = dist.metadata["Version"] or ""
             if name:
                 packages[name] = version
-    except Exception:
+    except Exception:  # noqa: S110 -- best-effort package collection; environment may lack importlib.metadata
```

Ruff (0.15.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)